src/actions/guides/frontend/internationalization.cr

```diff
@@ -0,0 +1,341 @@
+class Guides::Frontend::Internationalization < GuideAction
+  guide_route "/frontend/internationalization"
+
+  def self.title
+    "Internationalization - Handle Multiple Languages"
+  end
+
+  def markdown : String
+    <<-MD
+    This describes how to handle USERS with multiple language preferences.  At somepoint we can document how to add dynamic switching from the frontend too.
+
+    ## **Step 1** - add i18n shard to `shard.yml`
+
+    ```
+    dependencies:
+      i18n:
+        github: vladfaust/i18n.cr
+        version: ~> 0.1.1
+    ```
+    and of course install the shard
+    ```
+    shards install
+    ```
+
+    ## **Step 2** - add a starter localization and anothers needed:
+
+    first make a locales folder:
+    ```
+    mkdir config/locales
+    ```
+
+    then add a localization
+    ```
+    # config/locales/en.yml
+    en:
+      auth:
+        sign_in: "Sign-In"
+        sign_up: "Sign-up"
+        update_pwd: "Update Password"
+        reset_pwd_request: "Reset Password"
+      default:
+        page_name: Welcome
+      me:
+        email: Mail
+        profile: Profil
+        next: "Next, you may want to"
+        auth_guides: "Check out the authentication guides"
+        modify_page: "Modify this page"
+        after_signin: "Change where you go after sign in"
+    ```
+
+    ## **Step 3** - configure i18n to work with Lucky
+    ```
+    # config/i18n.cr
+    require "i18n"
+    require "i18n/backends/yaml"
+      I18n.backend = I18n::Backends::YAML.new.tap do |backend|
+      backend.load_paths << Dir.current + "/config/locales"
+      backend.load
+    end
+    ```
+
+    ## **Step 4** - create a user migration
+    generate a migration with:
+    ```
+    lucky db.migration AddLanguageToUser
+    ```
+
+    edit the file to look like (of course, the number will vary):
+    ```
+    # db/migrations/20191228100116_add_language_to_user.cr
+    class AddLanguageToUser::V20191228100116 < Avram::Migrator::Migration::V1
+      def migrate
+        alter table_for(User) do
+          add lang : String, default: "en"
+        end
+      end
+
+      def rollback
+        alter table_for(User) do
+          remove :lang
+        end
+      end
+    end
+    ```
+
+    and of course migrate
+    ```
+    lucky db.migrate
+    ```
+
+    ## **Step 5** - update the user model (add the new attribute)
+    ```
+    # src/models/user.cr
+      table do
+        column lang : String
+        column email : String
+        column encrypted_password : String
+       end
+    ```
+
+    ## **Step 6** - Update sign_up_user.cr (to validate the available languages) - ideally at somepoint in the future - lucky will autodect and autocreate the language list based on the config/locales yml files.
```

Comment:
I agree Lucky could autoload this but I think can be removed from the headline

Reply:
removed from headline - if you have tips on how to easily do that and add it to the validation and user drop-down - that would be a cool improvement to the doc